src/main/java/no/unit/nva/cristin/projects/CristinApiClient.java

```diff
@@ -94,9 +75,15 @@ public ProjectsWrapper queryCristinProjectsIntoWrapperObjectWithAdditionalMetada
         Map<String, String> requestQueryParams) throws ApiGatewayException {
 
         long startRequestTime = System.currentTimeMillis();
-        List<NvaProject> nvaProjects = importProjectsFromCristin(requestQueryParams);
+        HttpResponse<InputStream> response = queryProjects(requestQueryParams);
+        List<CristinProject> enrichedProjectsFromResponse =
+            getEnrichedProjectsUsingQueryResponse(response, requestQueryParams.get(LANGUAGE));
+        List<NvaProject> nvaProjects = mapValidCristinProjectsToNvaProjects(enrichedProjectsFromResponse);
         long endRequestTime = System.currentTimeMillis();
 
+        // TODO: NP-2385
+        //HttpHeaders headers = response.headers();
+
         ProjectsWrapper projectsWrapper = new ProjectsWrapper();
```

Comment:
This method is doing a lot, consider extracting the creation of the wrapper to a new method

Reply:
It is a separate issue to refactor building of a ProjectsWrapper object, so it will be addressed later